addon/attributes.js

```diff
@@ -20,6 +20,7 @@ function metaTypeFor(name, type, options) {
 
   if (options && options.polymorphic) {
     let typeKey = options.typeKey || 'type';
+    typeKey = typeof typeKey === 'function' ? '__dynamic__' : typeKey;
```

Comment:
multiline function breaks the regex used on metaType. That's why we are replacing it it by a static value